tailmq.go

```diff
@@ -1,154 +1,139 @@
+    "os"
+    "io/ioutil"
+    "net/url"
+    "strings"
+    "time"
+    "context"
+
+    "github.com/olaurendeau/tailmq/consumer"
+    "github.com/streadway/amqp"
 )
 
-var verbose bool
+const Help = `DESCRIPTION
+  TailMQ tail AMQP exchanges and output messages in stdout
+USAGE
+  tailmq [options] <exchange_name>
+EXAMPLES
+  tailmq amp.direct - Tail exchange amp.direct on local server with default access
+  tailmq -uri=amqp://user:password@tailmq.com:5672//awesome amp.topic - Tail exchange amp.topic from server tailmq.com in vhost /awesome
+  tailmq -server=prod amp.fanout - Tail exchange amp.fanout from server prod configured in file ~/.tailmq
+  tailmq -server=prod -vhost=/foobar amp.fanout - Tail exchange amp.fanout from server prod configured in file ~/.tailmq but use vhost /foobar
+OPTIONS
+`
+
+type Config struct {
+    uri *string
+    server *string
+    vhost *string
+    verbose *bool
+    prefix *bool
+    help *bool
+    exchangeName string
+    amqpConfigList []AmqpConfig
+}
 
 func main() {
 
-  var uri string
-  var exchangeName string
-  var prefix bool
-  
-  cli.VersionFlag = cli.BoolFlag{
-    Name: "version",
-    Usage: "print only the version",
-  }
+    ctx := context.Background()
 
-  app := cli.NewApp()
-  app.Name = "tailmq"
-  app.Usage = "Tail a RabbitMQ exchange"
-  app.Version = "0.1.0"
-  app.Compiled = time.Now()
-  app.ArgsUsage = "[exchangeName]"
-
-  app.Flags = []cli.Flag {
-
-    cli.StringFlag{
-      Name: "uri, u",
-      Value: "amqp://guest:guest@localhost:5672/",
-      Usage: "RabbitMQ amqp uri",
-      Destination: &uri,
-    },
-
-    cli.BoolFlag{
-      Name: "prefix",
-      Usage: "Should output be prefixed with date and time",
-      Destination: &prefix,
-    },
-
-    cli.BoolFlag{
-      Name: "verbose",
-      Usage: "Do you want more informations ?",
-      Destination: &verbose,
-    },
-  }
+    config := new(Config)
+
+    config.uri = flag.String("uri", "amqp://guest:guest@localhost:5672/", "RabbitMQ amqp uri")
+    config.server = flag.String("server", "", "Use predefined server from configuration")
+    config.vhost = flag.String("vhost", "", "Define vhost to tail from")
+    config.prefix = flag.Bool("prefix", false, "Should output be prefixed with datetime and time")
+    config.verbose = flag.Bool("verbose", false, "Do you want more informations ?")
+    config.help = flag.Bool("help", false, "How does it work ?")
+    flag.Parse()
 
-  app.Action = func(c *cli.Context) error {
+    config.amqpConfigList = getAmqpConfigList()
+    configureLogger(config)
+    displayHelp(config)
+    setUri(config)
+    
+    if (flag.NArg() == 0) {
+        log.Fatalf("Please choose an exchange to listen from")
+    } else if (flag.NArg() == 1) {
+        config.exchangeName = flag.Arg(0)
+    } else {
+        log.Fatalf("Not yet available")
+    }
 
-    if c.NArg() <= 0 {
-      log.Fatalf("Please choose an exchange to listen from")
+    c := consumer.New(*config.uri, config.exchangeName)
```

Comment:
I feel like I should pass the `ctx` struct to the consumer, but what should I do with it to get the `case <-ctx.Done(): return` useful ?